### PR TITLE
Added Karma testing env and Travis-CI support.

### PR DESCRIPTION
**Pull request by n1k0**
WiP. @leplatrem @natim @ametaireau @tarekziade, your feedback is welcome.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+language: node_js
```
Comment (almet) [inline .travis.yml:1]: not related, but setting sudo to false [will speedup build times by using docker](http://blog.travis-ci.com/2014-12-17-faster-builds-with-container-based-infrastructure/), see [the zamboni PR](https://github.com/mozilla/zamboni/pull/2821) for an example, or a [simpler one for the readinglist](https://github.com/mozilla-services/readinglist/commit/85dd328d2600909b72edbcb93d6c8194d01e9b5a).

File: karma.conf.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    logLevel: config.LOG_INFO,
+
+
+    // enable / disable watching file and executing tests whenever any file changes
+    autoWatch: false,
+
+
+    // start these browsers
+    // available browser launchers: https://npmjs.org/browse/keyword/karma-launcher
+    browsers: ['Firefox'],
```
Comment (almet) [inline karma.conf.js]: I'm not sure how long does this take to run, but at least on travis, we should aim to support multiple targets.

File: src/js/main.js [hunk 3]
```diff
@@ -0,0 +1,6 @@
+"use strict";
+
+var React = require("react");
+var App = require("./app");
+
+React.render(<App/>, document.querySelector("#app"));
```
Comment (almet) [inline src/js/main.js:6]: Wasn't aware of `querySelector`. nice.

**Comment by almet** [comment]
Pretty cool, I like that :)

File: karma.conf.js (hunk 2 above)
Comment (n1k0) [inline karma.conf.js]: Well for now it's a few seconds, mostly taken by startup/shutdown time; adding other browsers will add to that time, making running the suite a bit more painful for the developer :(

Maybe I can just setup some static page to run the tests, so the developer would use that locally, and use karma only for CI.

I'll do that.

Comment (n1k0) [inline karma.conf.js]: Tadaa.

screen shot 2015-01-27 at 08 45 47

**Comment by n1k0** [comment]
BAH. http://stackoverflow.com/a/19416096/330911

**Comment by n1k0** [comment]
OK I quit, Chrome support on Travis is painfully broken. Will switch to PhantomJS.

**Comment by n1k0** [comment]
Related Chrome/Travis issue: https://github.com/travis-ci/travis-ci/issues/2555

**Comment by n1k0** [comment]
Well, PhantomJS' JavascriptCore engine is too old for React. Discarding latest changes, will switch back to Firefox only CI for now and land this baby so we can move fwd.

**Comment by MoOx** [comment]
My 2 cents: testem is really simpler to configure than Karma.
I would say testem + saucelab + travis.

**Pull request closed without merging**